### ISSUE TITLE
Constructing function from closure using Magrittr results in unusable function

### ISSUE THREAD

**Issue by allswellthatsmaxwell** [smbache/magrittr#100]
When I use magrittr to pass a value for x to the function below, it results in an unusable function. I have version magrittr_1.5.

library(magrittr)
f <- function(x) { function(y) { x + y } }

# Case 1 (works)
f.5 <- f(5)
f.5(6) # returns 11

# Case 2 (works)
f.5 <- f(5) 
6 %>% f.5 # returns 11

# Case 3 (fails)
f.5 <- 5 %>% f
6 %>% f.5 # Error in x + y (from 1) : 
          # non-numeric argument to binary operator 

# Case 4 (fails)
f.5 <- 5 %>% f
f.5(6); # Same error as case 3

**Comment by smbache** [comment]
As mentioned by nongkron in this SO thread, it has something to do with `withVisible` and `freduce`. I'm not yet sure as to the reason for the behavior. "Plugging" `freduce` directly into the `pipe` code, rather than defining it separately in the package namespace is claimed to "fix" the problem. Not sure if it is the best/most satisfactory solution...

**Comment by smbache** [comment]
Revisiting the `freduce` function, I fail to remember a reason for the visibility checks; seems like a simpler definition is better:

```R
freduce <- function(value, function_list)
{
  k <- length(function_list)
  if (k == 1L)
    function_list[[1L]](value)
  else 
    Recall(function_list[[1L]](value), function_list[-1L])
}
```

And using this instead fixes the issue.

@hadley do you happen to know what reason I had ;-)   [I wouldn't think so]

**Comment by smbache** [comment]
fixed in 565eb9a6111fa432ab33284d13d59405621926a3
